rust/src/delta_datafusion.rs

```diff
@@ -163,9 +209,53 @@ impl TableProvider for delta::DeltaTable {
     }
 }
 
-fn stat_to_val(stat: &ColumnCountStat) -> Option<usize> {
-    match stat {
-        ColumnCountStat::Value(val) => Some(*val as usize),
-        ColumnCountStat::Column(_) => None,
+fn to_scalar_value(stat_val: &serde_json::Value) -> Option<datafusion::scalar::ScalarValue> {
```

Comment:
I think to be complete we have to pass the (arrow) datatype as well to this function, for example to parse/handle timestamps properly.

Reply:
Hm, for comparison purpose, I think using few basic primitive types are more easy. Maybe we can convert the compared `ScalarValue` to correct `ScalarValue` after comparison, e.g. `ScalarValue::Int64` to `ScalarValue::TimestampSecond` if it is Timestamp Second, etc.

Reply:
How about addressing this in another PR?

Reply:
Yeah perfectly fine to handle this in a follow up PR.

FYI, There was some other approach of using the aggregation code from DataFusion over here:
https://github.com/apache/arrow-datafusion/pull/719

Might also make sense to extend the ScalarValue a bit at DataFusion side  (e.g. derive `Ord`)

Reply:
+1 on extending ScalarValue in Datafusion wherever applicable for better code reuse. Implementing `Ord` trait there is a great idea.

Reply:
Sounds good, the suggestion. I will find some time to see if I can implement `Ord` there.